### ISSUE TITLE
Enable Autocomplete based on api for files with .xml extension

### ISSUE THREAD

**Issue by Solvaring** [pkulchenko/ZeroBraneStudio#789]
I created a long list of xml keywords that are relevant for making GUIs for Elder Scrolls Online. I pasted these into my plugin and classified them as type="keyword". Once the file extension becomes .xml all autocomplete from apis support is lost, so I'm forced to keep the list of keywords opened in the editor so that it can draw on that for autocomplete, or write the xml file with a .lua extension and then save it as .xml when i'm finished.

Please consider giving autocomplete based on api/plugin to .xml files and other files that don't end in a .lua extension.

**Comment by pkulchenko** [comment]
@Solvaring, how is the API registered?

**Comment by Solvaring** [comment]
So I just realized, I was able to get autocomplete to show up by changing the spec piece of my xml plugin i changed the apitype variable to apitype = "lua"

I also think I might be able to write a second api plugin and register it to the 'xml' group instead of group 'lua' because initially the apitype field of my spec was set to apitype="xml"

**Comment by pkulchenko** [comment]
@Solvaring, correct, the API usage depends on the apitype; that's why I asked how it's registered ;).

**Comment by pkulchenko** [comment]
@Solvaring, are you good with what we have so far or would you still need `AddAPI` call to add API that works with any interpreter/spec? Right now the only way to make it work is to use `api` config setting (https://studio.zerobrane.com/doc-general-preferences#custom-apis).

**Comment by Solvaring** [comment]
I think it's good the way i have it for now. I took the section of xml element tag keywords out of the main api table and moved it into its own unique table. In the spec part of the plugin i set ` exts = {"xml", "html"} ` and set ` apitype = "xml" ` in the onRegister event i registered the keywords table with

```
    table.insert(ide:GetConfig().api, name)
	table.insert(ide:GetConfig().api, "esoxmlkeywords")
	ide:AddAPI("lua", name, api)
	ide:AddAPI("xml", "esoxmlkeywords", esoxmlapi)
```

Now when im working in an xml file I get sane autocomplete suggestions from esoxmlapi instead of the main api table which holds all the lua stuff. When im working in a lua file i don't get autocomplete suggestions of xml keywords, so yeah everything is working nicely.
